### DISCUSSION TITLE
Pydantic Field's strict mode

### DISCUSSION THREAD

**Discussion by jonyscathe** [fastapi/sqlmodel#1115] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Optional

from sqlmodel import Field, Session, SQLModel, create_engine


class HeroBase(SQLModel):
    name: str = Field(index=True)
    secret_name: str
    age: int | None = Field(default=None, index=True, strict=True)


class Hero(HeroBase, table=True):
    id: int | None = Field(default=None, primary_key=True)


class HeroCreate(HeroBase):
    pass
```

### Description

I am currently refactoring some code.
Am trying to move from have separate use of pydantic, sqlalchemy & api to just using SQLModel.
My current pydantic classes use `strict=True` within the Fields to enforce strict mode rather than lax mode.
I had assumed that this behaviour would be available through the SQLModel definition by just adding `strict=True` into the Field.
And now that I look closer I see that there are several pydantic Field parameters that are not in the SQLModel Field such as strict, validate_default, init, init_var, kw_only.
Though the only one that I actually care about is strict.

I understand that if I add `model_config = ConfigDict(strict=True)` to the model then the entire model can become strict, however I want to make some integers and floats strict without making some date/datetime fields strict.

Wondering if this is possible at all?

### Operating System

Linux

### Operating System Details

python-slim docker container.

### SQLModel Version

0.0.22

### Python Version

3.12.6

**Comment by PierreFaniel** [comment]
I have encountered the same issue.
You should already be able to pass `strict=True` to the pydantic `Field` class via the `schema_extra` argument.

So in your example it would look like this:
```python
class HeroBase(SQLModel):
    ...
    age: int | None = Field(default=None, index=True, schema_extra={"strict": True})
```
